src/lib/homestore_backend/gc_manager.cpp

```diff
@@ -0,0 +1,360 @@
+
+    auto blk_size = homestore::data_service().get_blk_size();
+    auto shard_sb_size = sizeof(HSHomeObject::shard_info_superblk);
+    auto total_size = sisl::round_up(shard_sb_size, blk_size);
+    auto shard_sb_buf = iomanager.iobuf_alloc(blk_size, total_size);
+
+    std::memcpy(shard_sb_buf, shard_sb, shard_sb_size);
+
+    sisl::sg_list shard_sb_sgs;
+    shard_sb_sgs.size = total_size;
+    shard_sb_sgs.iovs.emplace_back(iovec{.iov_base = shard_sb_buf, .iov_len = total_size});
+    return shard_sb_sgs;
+}
+
+bool GCManager::pdev_gc_actor::copy_valid_data(chunk_id_t move_from_chunk, chunk_id_t move_to_chunk, bool is_emergent) {
+    return true;
+}
+
+void GCManager::pdev_gc_actor::purge_reserved_chunk(chunk_id_t chunk) {
+    auto vchunk = m_chunk_selector->get_extend_vchunk(chunk);
+    RELEASE_ASSERT(!vchunk->m_pg_id.has_value(), "chunk_id={} is expected to a reserved chunk, and not belong to a pg",
+                   chunk);
+    vchunk->reset(); // reset the chunk to make sure it is empty
+
+    // clear all the entries of this chunk in the gc index table
+    auto start_key = BlobRouteByChunkKey{BlobRouteByChunk(chunk, 0, 0)};
+    auto end_key = BlobRouteByChunkKey{
+        BlobRouteByChunk{chunk, std::numeric_limits< uint64_t >::max(), std::numeric_limits< uint64_t >::max()}};
+    homestore::BtreeRangeRemoveRequest< BlobRouteByChunkKey > range_remove_req{
+        homestore::BtreeKeyRange< BlobRouteByChunkKey >{
+            std::move(start_key), true /* inclusive */, std::move(end_key), true /* inclusive */
+        }};
+    auto status = m_index_table->remove(range_remove_req);
+    if (status != homestore::btree_status_t::success) {
+        // TODO:: handle the error case here!
+        RELEASE_ASSERT(false, "can not clear gc index table, chunk_id={}", chunk);
+    }
+}
+
+void GCManager::pdev_gc_actor::process_gc_task(gc_task& task) {
+    auto priority = task.get_priority();
+    auto move_from_chunk = task.get_move_from_chunk_id();
+    auto move_from_vchunk = m_chunk_selector->get_extend_vchunk(move_from_chunk);
+
+    LOGINFO("start process gc task for move_from_chunk={} with priority={} ", move_from_chunk, priority);
+
+    // we need to select the move_from_chunk out of the per pg chunk heap in chunk selector if it is a gc task with
+    // normal priority. for the task with emergent priority, it is already selected since it is now used for an open
+    // shard.
+
+    if (!GCManager::is_eligible_for_gc(move_from_vchunk)) {
+        LOGWARN("move_from_chunk={} is not eligible for gc, so we can not process the gc task", move_from_chunk);
+        task.complete(false);
+        return;
+    }
+
+    LOGINFO("move_from_chunk={} belongs to pg {} ", move_from_chunk, move_from_vchunk->m_pg_id.value());
+
+    // make chunk to gc state, so that it can be select for creating shard
+    auto succeed = m_chunk_selector->try_mark_chunk_to_gc_state(
```

Comment:
@xiaoxichen I now use a new state `gc` for vchunk, so that it will not be selected for creating shard